samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -19,36 +19,195 @@
+      LocalProcessorListener listener = new LocalProcessorListener();
+      plan.getJobConfigs().forEach(jobConfig -> {
+          log.info("Starting job {} StreamProcessor with config {}", jobConfig.getName(), jobConfig);
+          StreamProcessor processor = createStreamProcessor(jobConfig, app, listener);
+          processor.start();
+          processors.add(processor);
+        });
+      appStatus = ApplicationStatus.Running;
+
+      // 4. block until the processors are done or there is a failure
+      awaitComplete();
+
+    } catch (Throwable t) {
+      appStatus = ApplicationStatus.UnsuccessfulFinish;
+      throw new SamzaException("Failed to run application", t);
+    } finally {
+      if (coordination != null) {
+        coordination.reset();
+      }
+    }
   }
 
   @Override
   public void kill(StreamApplication streamApp) {
+    processors.forEach(StreamProcessor::stop);
   }
 
   @Override
   public ApplicationStatus status(StreamApplication streamApp) {
-    return null;
+    return appStatus;
+  }
+
+  /**
+   * Create the Coordination needed by the application runner.
+   * @return {@link CoordinationUtils}
+   * @throws Exception exception
+   */
+  /* package private */ CoordinationUtils getCoordinationUtils() throws Exception {
+    ApplicationConfig appConfig = new ApplicationConfig(config);
+    String clazz = appConfig.getCoordinationServiceFactoryClass();
+    if (clazz != null) {
+      CoordinationServiceFactory factory = ClassLoaderHelper.fromClassName(clazz);
+      String groupId = String.format("app-%s-%s", appConfig.getAppName(), appConfig.getAppId());
+      return factory.getCoordinationService(groupId, uid, config);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Create intermediate streams.
+   * @param intStreams intermediate {@link StreamSpec}s
+   * @throws Exception exception
+   */
+  /* package private */ void createStreams(List<StreamSpec> intStreams) throws Exception {
+    if (!intStreams.isEmpty()) {
+      if (coordination != null) {
+        Latch initLatch = coordination.getLatch(1, LATCH_INIT);
+        coordination.getLeaderElector().tryBecomeLeader(() -> {
+            getStreamManager().createStreams(intStreams);
+            initLatch.countDown();
+          });
+        initLatch.await(LATCH_TIMEOUT_MINUTES, TimeUnit.MINUTES);
+      } else {
+        // each application process will try creating the streams, which
+        // requires stream creation to be idempotent
+        getStreamManager().createStreams(intStreams);
+      }
+    }
+  }
+
+  /**
+   * Create {@link StreamProcessor} based on {@link StreamApplication} and the config
+   * @param config config
+   * @param app {@link StreamApplication}
+   * @return {@link StreamProcessor]}
+   */
+  /* package private */ StreamProcessor createStreamProcessor(Config config, StreamApplication app, StreamProcessorLifeCycleAware listener) {
+    Object taskFactory = TaskFactoryUtil.createTaskFactory(config, app, this);
```

Comment:
Shouldn't this be a constructor in StreamProcessor instead of a helper here?
Btw, @navina, Is StreamProcessor going to be a framework internal class in future?